tubular/nominal.py

```diff
@@ -627,6 +618,62 @@ def fit(self, X, y):
                     group_means, group_weight
                 ).to_dict()
 
+
+    def fit(self, X, y):
+        """Identify mapping of categorical levels to mean response values.
+
+        If the user specified the weights_column arg in when initialising the transformer
+        the weighted mean response will be calculated using that column.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to with catgeorical variable columns to transform and also containing response_column
+            column.
+
+        y : pd.Series
+            Response variable or target.
+
+        """
+
+        BaseNominalTransformer.fit(self, X, y)
+
+        self.mappings = {}
+
+        if self.level:
+
+            if self.level == 'all':
+
+                self.response_levels = y.unique()
+
+            else:
+
+                if isinstance(self.level, str):
+                    self.level = [self.level]
+
+                if any([level not in list(y.unique()) for level in self.level]):
+                    raise ValueError("Levels contains a level to encode against that is not present in the response.")
+
+                self.response_levels = self.level
+                
+            self.transformer_dict = {}
+            mapped_columns = []
+
+            for level in self.response_levels:
+                columns = [column + '_' + level for column in self.columns]
+                y_temp = y.apply(lambda x: x == level)
```

Comment:
if we create y_temp for each level do we need to pass a level to fit_binary_response?

Reply:
Passing level is to control the generation of new columns in (X) temp in _fit_binary_response - I could probably change how this works, although I can't see immediately how without creating an X_temp before the method call

Reply:
Done in this commit, unfortunately I can't see a way of removing the columns argument to make it just of the form fit(X,y). There's maybe a way involving an e.g. dictionary attribute that stores the new columns for each level, but I think that telling fit which element to access would be pretty convoluted.

https://github.com/lvgig/tubular/pull/67/commits/bf53838f22cb21e17ba1514d4a95c7eadda68e9a

Reply:
I think it would also work to take a single column at a time and iterate over columns for line 663, but I'm happy enough with the approach taken here.